drivers/firmware/scmi/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+# SCMI core files
+zephyr_library_sources_ifdef(CONFIG_ARM_SCMI core.c)
+zephyr_library_sources_ifdef(CONFIG_ARM_SCMI_MAILBOX_TRANSPORT mailbox.c)
+zephyr_library_sources_ifdef(CONFIG_ARM_SCMI_MAILBOX_TRANSPORT shmem.c)
```

Comment:
This should either be split into two configs; one for mailbox, one for shmem, or, combine them like
```
zephyr_library_sources_ifdef(CONFIG_ARM_SCMI_MAILBOX_TRANSPORT mailbox.c shmem.c)
```

Reply:
edit, please do split them, they both have separate init priorities, so they are seemingly separate features :) if they depend on each other, use Kconfig to manage this